Sources/Datadog/Tracing/Span/SpanBuilder.swift

```diff
@@ -23,50 +23,129 @@ internal struct SpanBuilder {
-        }
+        tags = baggageItems
 
-        // Add regular tags
-        for (tagName, tagValue) in tagsReducer.reducedSpanTags {
-            jsonStringEncodedTags[tagName] = JSONStringEncodableValue(tagValue, encodedUsing: tagsJSONEncoder)
-        }
+        // Add regular tags (prefer regular tags over baggate items)
+        let regularTags = castValuesToString(tagsReducer.reducedSpanTags)
+        tags.merge(regularTags) { _, regularTag in regularTag }
 
         // Transform user info to `Span.UserInfo` representation
         let userInfo = userInfoProvider.value
         let spanUserInfo = Span.UserInfo(
             id: userInfo.id,
             name: userInfo.name,
             email: userInfo.email,
-            extraInfo: userInfo.extraInfo.mapValues { value in JSONStringEncodableValue(value, encodedUsing: tagsJSONEncoder) }
+            extraInfo: castValuesToString(userInfo.extraInfo)
         )
 
-        return Span(
-            traceID: ddspan.ddContext.traceID,
-            spanID: ddspan.ddContext.spanID,
-            parentID: ddspan.ddContext.parentSpanID,
-            operationName: ddspan.operationName,
+        let span = Span(
+            traceID: traceID,
+            spanID: spanID,
+            parentID: parentSpanID,
+            operationName: operationName,
             serviceName: serviceName,
-            resource: tagsReducer.extractedResourceName ?? ddspan.operationName,
-            startTime: dateCorrector.currentCorrection.applying(to: ddspan.startTime),
-            duration: finishTime.timeIntervalSince(ddspan.startTime),
+            resource: tagsReducer.extractedResourceName ?? operationName,
+            startTime: dateCorrector.currentCorrection.applying(to: startTime),
+            duration: finishTime.timeIntervalSince(startTime),
             isError: tagsReducer.extractedIsError ?? false,
             source: source,
             tracerVersion: sdkVersion,
             applicationVersion: applicationVersion,
             networkConnectionInfo: networkConnectionInfoProvider?.current,
             mobileCarrierInfo: carrierInfoProvider?.current,
             userInfo: spanUserInfo,
-            tags: jsonStringEncodedTags
+            tags: tags
         )
+
+        return span
+    }
+
+    // MARK: - Attributes Conversion
+
+    /// Encodes `Span` attributes to JSON strings
+    private let attributesJSONEncoder: JSONEncoder = .default()
+
+    /// Converts `Encodable` attributes to its lossless JSON string representation, e.g.:
+    /// * it will convert `"abc"` string value to `"abc"` JSON string value
+    /// * it will convert `1` integer value to `"1"` JSON string value
+    /// * it will convert `true` boolean value to `"true"` JSON string value
+    /// * it will convert `Person(name: "foo")` encodable struct to `"{\"name\": \"foo\"}"` JSON string value
+    private func castValuesToString(_ dictionary: [String: Encodable]) -> [String: String] {
```

Comment:
💡  This is the old code from `JSONStringEncodableValue` it's just moved to `SpanBuilder` as now we convert `[String: Encodable]` to `[String: String]` in there, instead of doing this lazy on writer's thread.